nilearn/datasets/func.py

```diff
@@ -1276,3 +1276,174 @@ def fetch_mixed_gambles(n_subjects=1, data_dir=None, url=None, resume=True,
+    Register and sign the Open Access Data Use Terms at
+    ConnectomeDB: https://db.humanconnectome.org/
+    """
+    # Data is manually uploaded in NITRC
+    url = "https://www.nitrc.org/frs/download.php/8037/Megatrawls.tgz"
+    opts = {'uncompress': True}
+
+    # dataset terms
+    dimensionalities = ['d25', 'd50', 'd100', 'd200', 'd300']
+    timeseries_methods = ['ts2', 'ts3']
+
+    message = ("The %s you have given '%s' is invalid. Please choose either "
+               "of them %s or list of specific choices.")
+
+    names = ['choice_dimensionality', 'choice_timeseries']
+    inputs = [choice_dimensionality, choice_timeseries]
+    assign_names = ['dimensionalities', 'timeseries_methods']
+    standard_inputs = [dimensionalities, timeseries_methods]
+
+    for name_, input_, assign_, standard_ in zip(names, inputs, assign_names,
+                                                 standard_inputs):
+        if input_ is not None:
+            if isinstance(input_, list):
+                for n_input_ in input_:
+                    if n_input_ not in standard_:
+                        raise ValueError(message % (
+                            name_, input_, str(standard_)))
+                if assign_ == 'dimensionalities':
+                    dimensionalities = input_
+                else:
+                    timeseries_methods = input_
+            elif not isinstance(input_, list):
+                if input_ not in standard_:
+                    raise ValueError(message % (
+                        name_, input_, str(standard_)))
+
+                if assign_ == 'dimensionalities':
+                    dimensionalities = [input_]
+                else:
+                    timeseries_methods = [input_]
+
+    n_combinations = len(dimensionalities) * len(timeseries_methods)
+    dataset_name = 'Megatrawls'
+
+    files_netmats1 = []
+    files_netmats2 = []
+    for dim_ in dimensionalities:
+        filename_ = os.path.join('3T_Q1-Q6related468_MSMsulc_' + str(dim_))
+        for timeserie_ in timeseries_methods:
+            each_files_netmats1 = [(os.path.join(
+                filename_ + '_' + str(timeserie_), 'Znet1.txt'), url, opts)]
+            each_files_netmats2 = [(os.path.join(
+                filename_ + '_' + str(timeserie_), 'Znet2.txt'), url, opts)]
+            files_netmats1.append(each_files_netmats1)
+            files_netmats2.append(each_files_netmats2)
+
+    data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir, verbose=verbose)
+    network_matrices1 = []
+    network_matrices2 = []
+    for n_ in range(n_combinations):
+        netmats1 = _fetch_files(
+            data_dir, files_netmats1[n_], resume=resume, verbose=verbose)
+        network_matrices1.extend(netmats1)
+        netmats2 = _fetch_files(
+            data_dir, files_netmats2[n_], resume=resume, verbose=verbose)
+        network_matrices2.extend(netmats2)
+
+    Znet1 = network_matrices1
+    Znet2 = network_matrices2
```

Comment:
Can we not stick to returning path of the text files? I see confounds in some fetchers are returned the same.